internal/statement_parser.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
```

Comment:
I wonder can we use spansql for this parsing?

Reply:
No, I'm afraid that would cause a lot of problems, as spansql has a slightly different intended usage:
1. The parsing in this library is only interested in extracting some very specific parts of the sql string, and will ignore everything else. That means that if the SQL syntax is extended with new features, the parsing in this library will still work and just ignore that.
2. The parsing in spansql is intended to parse and understand the entire sql string. Any sql string that it does not understand will cause an error. There are still multiple existing features in the SQL syntax that are not understood by spansql, and any future additions to the syntax must be added to spansql for the parsing to accept them.

Reply:
Okay. Thanks for the explanation.